src/wasm/wasm-ir-builder.cpp

```diff
@@ -1523,6 +1523,13 @@ Result<> IRBuilder::makeTableCopy(Name destTable, Name srcTable) {
   return Ok{};
 }
 
+Result<> IRBuilder::makeTableInit(Name elem, Name table) {
+  TableInit curr;
+  CHECK_ERR(visitTableInit(&curr));
```

Comment:
This is going to need the table to be set on `curr` so it can figure out whether to read an i32 or an i64. Since we're not using child-typer in the validator yet, the only way to test that this is correct is by seeing how the instruction is parsed when one of its subsequent children is unreachable.

Reply:
I don't follow. There is no implementation of `visitTableInit` - it's empty, then, from the parent of IRBuilder, which is UnifiedExpressionVisitor. That calls visitExpression, but there is no special code from TableInit there either. Am I mixed up..?

If this is some table64 complexity, then perhaps it is simpler if I make this PR only support table.init on wasm32?

Reply:
`visitTableInit` ends up in `visitExpression`, which calls `ConstraintCollector::visit`. `ConstraintCollector` inherits from `ChildTyper`, which should implement a specific `visitTableInit` that will need to look up the table to figure out the index type.

But yes, this is to properly support table64, so one option would be to punt that to later. I think it would be better to handle it now, though.

Reply:
Ah, thanks, it is indeed in that other class. Done.